### PR TITLE
Added authentication before retreiving data from github

### PR DESCRIPTION
**Pull request by GeeH**

### CODE REVIEW

**Comment by Ocramius** [comment]
Will give this one a shot in few minutes.

**Comment by ins0** [comment]
this solves #288 

@Ocramius this pr should be merged as well #304 as the github client throws exceptions for 404 repository content

**Comment by Ocramius** [comment]
@ins0 I told @GeeH to provide a cleaner solution: he's doing that right now.

**Comment by ins0** [comment]
would be nice to get informed as i asked you on irc if the implementation in pr #304 is ok and u agreed so i close 304?

File: module/ZfModule/src/ZfModule/Delegators/EdpGithubClientAuthenticator.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+use EdpGithub\Client;
+use Zend\ServiceManager\DelegatorFactoryInterface;
+use Zend\ServiceManager\ServiceLocatorInterface;
+
+class EdpGithubClientAuthenticator implements DelegatorFactoryInterface
+{
+
+    /**
+     * A factory that creates delegates of a given service
```
Comment (Ocramius) [inline module/ZfModule/src/ZfModule/Delegators/EdpGithubClientAuthenticator.php]: all zis stuff, except for `@return` is `{@inheritDoc}`

File: module/ZfModule/src/ZfModule/Delegators/EdpGithubClientAuthenticator.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<?php
+/**
```
Comment (Ocramius) [inline module/ZfModule/src/ZfModule/Delegators/EdpGithubClientAuthenticator.php]: Fix headers

File: module/ZfModule/src/ZfModule/Delegators/EdpGithubClientAuthenticator.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    public function createDelegatorWithName(ServiceLocatorInterface $serviceLocator, $name, $requestedName, $callback)
+    {
+        /** @var Client $client */
+        $client = $callback();
+        $config = $serviceLocator->get('config')['scn-social-auth'];
+        $client->authenticate('UrlClientId', $config['github_client_id'], $config['github_secret']);
+
+        return $client;
+    }
+}
```
Comment (Ocramius) [inline module/ZfModule/src/ZfModule/Delegators/EdpGithubClientAuthenticator.php]: EOF EOL

File: module/ZfModule/src/ZfModule/Delegators/EdpGithubClientAuthenticator.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+     * @param ServiceLocatorInterface $serviceLocator the service locator which requested the service
+     * @param string $name the normalized service name
+     * @param string $requestedName the requested service name
+     * @param callable $callback the callback that is responsible for creating the service
+     *
+     * @return Client
+     */
+    public function createDelegatorWithName(ServiceLocatorInterface $serviceLocator, $name, $requestedName, $callback)
+    {
+        /** @var Client $client */
```
Comment (Ocramius) [inline module/ZfModule/src/ZfModule/Delegators/EdpGithubClientAuthenticator.php]: `/*`, not `/**`

**Pull request merged**